presto-main/src/main/java/io/prestosql/operator/DynamicFilterSourceOperator.java

```diff
@@ -129,34 +137,48 @@ public OperatorFactory duplicate()
     private final Consumer<TupleDomain<DynamicFilterId>> dynamicPredicateConsumer;
     private final int maxFilterPositionsCount;
     private final long maxFilterSizeInBytes;
+    private int minMaxCollectionLimit;
 
     private final List<Channel> channels;
+    private final List<Integer> minMaxChannels;
 
     // May be dropped if the predicate becomes too large.
     @Nullable
     private BlockBuilder[] blockBuilders;
     @Nullable
     private TypedSet[] valueSets;
 
+    @Nullable
+    private Block[] minValues;
+    @Nullable
+    private Block[] maxValues;
+
     private DynamicFilterSourceOperator(
             OperatorContext context,
             Consumer<TupleDomain<DynamicFilterId>> dynamicPredicateConsumer,
             List<Channel> channels,
             PlanNodeId planNodeId,
             int maxFilterPositionsCount,
-            DataSize maxFilterSize)
+            DataSize maxFilterSize,
+            int minMaxCollectionLimit)
     {
         this.context = requireNonNull(context, "context is null");
         this.maxFilterPositionsCount = maxFilterPositionsCount;
         this.maxFilterSizeInBytes = maxFilterSize.toBytes();
+        this.minMaxCollectionLimit = minMaxCollectionLimit;
 
         this.dynamicPredicateConsumer = requireNonNull(dynamicPredicateConsumer, "dynamicPredicateConsumer is null");
         this.channels = requireNonNull(channels, "channels is null");
 
         this.blockBuilders = new BlockBuilder[channels.size()];
         this.valueSets = new TypedSet[channels.size()];
+        ImmutableList.Builder<Integer> minMaxChannelsBuilder = ImmutableList.builder();
         for (int channelIndex = 0; channelIndex < channels.size(); ++channelIndex) {
             Type type = channels.get(channelIndex).type;
+            // Skipping DOUBLE and REAL in collectMinMaxValues to avoid dealing with NaN values
+            if (minMaxCollectionLimit > 0 && type.isOrderable() && type != DOUBLE && type != REAL) {
```

Comment:
can we add support for `DOUBLE` and `REAL` by filtering `NaN` at the end of min/max collection?